packages/ripgrep-workspace/src/node/ripgrep-process-module.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { ContainerModule, Container } from 'inversify';
+import { ProcessManager } from "@theia/process/lib/node";
+import { ILogger } from '@theia/core/lib/common/logger';
+import { RipGrepWorkSpace, RipGrepProcessOptions, RipGrepProcessFactory } from './ripgrep-workspace';
+
+export const ripgrepProcessModule = new ContainerModule(bind => {
+    bind(RipGrepWorkSpace).toSelf().inTransientScope();
+    bind(RipGrepProcessFactory).toFactory(ctx =>
+        (options: RipGrepProcessOptions) => {
+            const child = new Container({ defaultScope: 'Singleton' });
+            child.parent = ctx.container;
+            const logger = ctx.container.get<ILogger>(ILogger);
+            const loggerChild = logger.child({ 'module': 'ripgrepProcess' });
```

Comment:
we should do this with 1 logger as a singleton per package see terminal-backend-module.ts